notebook/tests/selenium/test_display_id.py

```diff
@@ -0,0 +1,160 @@
+"""
+Various output tests.
+"""
+import pytest
+import textwrap
+
+from .utils import wait_for_script_to_return_true
+
+INITIAL_CELLS = [textwrap.dedent("""\
+    ip = get_ipython()
+    from IPython.display import display
+    def display_with_id(obj, display_id, update=False, execute_result=False):
+        iopub = ip.kernel.iopub_socket
+        session = get_ipython().kernel.session
+        data, md = ip.display_formatter.format(obj)
+        transient = {'display_id': display_id}
+        content = {'data': data, 'metadata': md, 'transient': transient}
+        if execute_result:
+            msg_type = 'execute_result'
+            content['execution_count'] = ip.execution_count
+        else:
+            msg_type = 'update_display_data' if update else 'display_data'
+        session.send(iopub, msg_type, content, parent=ip.parent_header)
+    """)]
+
+
+@pytest.fixture
+def get_outputs(notebook):
+    def _get_outputs(cell_idx):
+        return notebook.browser.execute_script(f"""\
+            var cell = Jupyter.notebook.get_cell({cell_idx});
+            return cell.output_area.outputs;
+            """)
+    return _get_outputs
+
+
+@pytest.fixture
+def get_iopub_messages(notebook):
+    def _get_iopub_messages(cell_idx):
+        script = f"return IPython.notebook.get_cell({cell_idx}).iopub_messages"
+        return notebook.browser.execute_script(script)
+    return _get_iopub_messages
+
+
+def test_display_id(prefill_notebook, get_outputs, get_iopub_messages):
+    notebook = prefill_notebook(INITIAL_CELLS)
+    notebook.execute_cell(0)
+
+    #TODO describe test 1
```

Comment:
Since this is all one big, long test, it'd be nice to have a short description of what each part is testing. I'm not really sure what it _is_ that is being tested here. Could someone who knows more maybe suggest a short summary?